### PR TITLE
Run HIP CI with -Werror

### PR DESCRIPTION
**Pull request by masterleinad**
With `-Wno-unused-command-line-argument` and `-Wno-braced-scalar-init` we should be able to also check for warnings in the HIP CI tester.

### CODE REVIEW

File: containers/src/Kokkos_ScatterView.hpp [hunk 1]
```diff
@@ -558,6 +558,8 @@ struct ReduceDuplicatesBase {
       Kokkos::Profiling::beginParallelFor(std::string("reduce_") + name, 0,
                                           &kpID);
     }
+#else
+    (void)name;
```
Comment (masterleinad) [inline containers/src/Kokkos_ScatterView.hpp]: Apparently, we never tried to compile without profiling in any of the other regression testers.

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 2]
```diff
@@ -641,21 +641,25 @@ struct SubviewExtents {
     error(buf + n, buf_len - n, domain_rank + 1, range_rank + 1, dim, args...);
   }
 
+#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
   template <size_t... DimArgs, class... Args>
   KOKKOS_FORCEINLINE_FUNCTION void error(const ViewDimension<DimArgs...>& dim,
                                          Args... args) const {
-#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
```
Comment (dalg24) [inline core/src/impl/Kokkos_ViewMapping.hpp]: Why did you change this?

Comment (masterleinad) [reply core/src/impl/Kokkos_ViewMapping.hpp]: If `KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST` is not defined, `dim` and `args...` are not used. Since the `(void)` trick doesn't work on parameter packs, this seemed to be the best solution to avoid the warning.

**Review by crtrott — APPROVED** [review]
